workflows/snvphylnfc.nf

```diff
@@ -84,21 +84,39 @@ workflow SNVPHYL {
 
     ch_versions = Channel.empty()
 
+    // Track processed IDs
+    def processedIDs = [] as Set
+
     // Create a new channel of metadata from a sample sheet
     // NB: `input` corresponds to `params.input` and associated sample sheet schema
     input = Channel.fromSamplesheet("input")
         // Map the inputs so that they conform to the nf-core-expected "reads" format.
         // Either [meta, [fastq_1], reference_assembly]
         // or [meta, [fastq_1, fastq_2], reference_assembly] if fastq_2 exists
+        // and remove non-alphanumeric characters in sample_names (meta.id), whilst also correcting for duplicate sample_names (meta.id)
         .map { meta, fastq_1, fastq_2, reference_assembly ->
+            if (!meta.id) {
+                meta.id = meta.irida_id
+            } else {
+                // Replace spaces in 'id' with underscores
+                meta.id = meta.id.replaceAll(/[^A-Za-z0-9_.\-]/, '_')
+            }
```

Comment:
Should the second part be in an else block? Do we not want it to run if we do `meta.id = meta.irida_id` first?

Reply:
`meta.irida_id` will assigned by IRIDA-Next (user cannot modify this) when run in IRIDA-Next, and thus we do not need to worry about code injection. For `meta.id` we need to be a bit more careful because a user can supply this value. If someone runs the pipeline locally the `schema_input.json` should be sufficient for `meta.irida_id`. I should however update the comment as it does more than just remove spaces now!

Reply:
1b3dee5